main.py

```diff
@@ -12,6 +12,52 @@
 
 # Import whatever libraries/modules you need
 
-import numpy as np
+import tensorflow as tf
+from tensorflow.keras.datasets import cifar10
+from tensorflow.keras.preprocessing.image import ImageDataGenerator
+from tensorflow.keras.models import Sequential
+from tensorflow.keras.layers import Dense, Dropout, Activation, Flatten
+from tensorflow.keras.layers import Conv2D, MaxPooling2D
+cifar10 = tf.keras.datasets.cifar10 
+model = tf.keras.models.Sequential() 
 
-# Your working code here
+def data_preparation():
+    (x_train, y_train), (x_test, y_test) = cifar10.load_data()
+
+    x_train = tf.keras.utils.normalize(x_train, axis=1)
+    x_test = tf.keras.utils.normalize(x_test, axis=1)
+
+def model_construction():
+    # First convolution layer
+    model.add(Conv2D(256, (3, 3), input_shape=x_train.shape[1:]))
+    model.add(Activation('relu'))
+    model.add(MaxPooling2D(pool_size=(2,2)))
+
+    # second convolution layer
+    model.add(Conv2D(256, (3, 3)))
+    model.add(Activation('relu'))
+    model.add(MaxPooling2D(pool_size=(2,2)))
+
+    model.add(Flatten())  # this converts our 3D feature maps to 1D feature vectors
+    model.add(Dense(64))
+
+    # output layer
+    model.add(Dense(10))
+    model.add(Activation('softmax'))
+
+    model.compile(optimizer='adam', # default oprimizer
+                loss='sparse_categorical_crossentropy', 
+                metrics=['accuracy']) #keep track of accuracy along the way
+
+def model_training():
+    model.fit(x_train, y_train, epochs=20)
+
+def model_validation():
+    val_loss, val_acc = model.evaluate(x_test, y_test) #validate model
+    print(val_loss, val_acc)
+
+if __name__ == "__main__":
+    data_preparation()
```

Comment:
This is not necessarily the best way to use these functions, but it's nothing that bad.  It's how I expect someone not as familiar with fp to write functions. Ideally though, you want functions to return and not affect any variables outside the function, this isn't possible sometimes but it's good to avoid it.

So the best way to do this would be: 

``` 
data_preperation():
   ... do stuff
   return (x_train, y_train), (x_test, y_test)

def model_construction():
   ... do stuff
   return model

def train_model(model, x_train, y_train):
    .... train model
   # No return here, so this function has side effects

def model_validation(model):
     .... evaluate model
     return val_loss, val_acc

if __name__ == "__main__":
    (x_train, y_train), (x_test, y_test) = data_preparation()
    model = model_construction()
    model_training(model)
    val_loss, val_acc = model_validation(model)
```
Anyways that's just me being an fp nerd. Well done otherwise